MediaProcessor/src/FFmpegConfigManager.h

```diff
@@ -20,27 +25,56 @@ enum class VideoCodec { H264, H265, VP8, VP9, UNKNOWN };
  */
 class FFmpegConfigManager {
    public:
+    struct FFmpegGlobalSettings {
+        bool overwrite = false;
+        CodecStrictness strictness = CodecStrictness::EXPERIMENTAL;
+        fs::path inputFilePath;
+        fs::path outputFilePath;
+    } m_globalSettings;
+
+    struct FFmpegAudioSettings {
+        AudioCodec codec = AudioCodec::AAC;
+        fs::path chunksPath;
+        std::vector<fs::path> processedChunkCol;
+        int numChunks;
+        int sampleRate = 48000;
+        int numChannels = 2;
+        double overlapDuration = DEFAULT_OVERLAP_DURATION;
+    } m_audioSettings;
+
+    struct FFmpegVideoSettings {
+        VideoCodec codec = VideoCodec::H264;
+    } m_videoSettings;
+
```

Comment:
I think I share your opinion on the separation between the two classes. I had only relocated the variables for ease of access (especially for the tests), but I do agree they're not really "config" options. 

I'll get rid of `int numChunks` and `double overlapDuration`, what do you think about `fs::path chunksPath` and `std::vector<fs::path> processedChunkCol`? Personally I see the chunks path as redundant (that is, I should remove it), while the other arguably belongs to both **but** maybe more on the side of AudioProcessor?

I'm also curious how you feel about my moving methods away from AudioProcessor to the controller. If you feel like they are better suited to AudioProcessor I have no problems moving them back.

Re: the structs - it was intentional - with updateSettings taking the structs rather than individual options won't we have to make them public so the user can initialize them? Sorry if I'm missing something.

Reply:
> I'll get rid of int numChunks and double overlapDuration, what do you think about fs::path chunksPath and std::vector<fs::path> processedChunkCol? Personally I see the chunks path as redundant (that is, I should remove it), while the other arguably belongs to both but maybe more on the side of AudioProcessor?

I think the latter is also something the AudioProcessor should own and manage. Thinking from the FFmpeg side, we shouldn't necessarily be concerned with if the input is in chunks or a full file, etc. Those are AudioProcessor's duties. This also would eliminate the redundancy as you said. I do see it will make testing a little less convenient, but in terms of design I feel it would be more clear. So, instead of holding copies of strictly audio-related information, we should receive them where relevant as arguments. Let me know if you see something not feasible with this though.  

> I'm also curious how you feel about my moving methods away from AudioProcessor to the controller. If you feel like they are better suited to AudioProcessor I have no problems moving them back.

Any functionality that uses FFmpeg should reside in the FFmpegController. So, I think that was a good decision.

> Re: the structs - it was intentional - with updateSettings taking the structs rather than individual options won't we have to make them public so the user can initialize them? Sorry if I'm missing something.

You're right, sure. I'm still keen on keeping the structs private so for now we can make the controller a friend as the two are instrinsically coupled anyway. 

Thank you and please let me know if I can help in any way!